test/unit/spec.js

```diff
@@ -37,3 +37,15 @@ Object.keys(spec).map(function(folder) {
     document.write(string.join(''));
   });
 });
+
+p5._throwValidationErrors = true;
+
+// a custom assertion for validation errors that correctly handles
+// minified p5 libraries.
+assert.validationError = function(fn) {
+  if (p5.ValidationError) {
+    assert.throws(fn, p5.ValidationError);
+  } else {
+    assert.doesNotThrow(fn, Error, 'got unwanted exception');
+  }
+};
```

Comment:
I don't really think this should be in the `spec.js` file. I was looking at adding some helpers in too and I think the best course of action is probably to add a `p5_helpers.js` file to the `test/js/` folder and then add it in the test html files, similar to how `chai_helpers.js` is added.

Reply:
Ok... I know very little about mocha or how it's files are supposed to be structured. This is the first place I found that worked for both the in-browser and command-line execution of the tests. Requiring a per-suite setting of this global flag seems a little fragile.

Reply:
Ok... I know very little about mocha or how it's files are supposed to be structured. This is the first place I found that worked for both the in-browser and command-line execution of the tests. Requiring a per-suite setting of this global flag seems a little fragile.

Reply:
Yeah, it is a bit fragile - really the code in the html files should be factored out into a js file at least, but it hasn't been done yet. I see there's remnants of you attempting to set the `p5._throwValidtionErrors` in those files in the diffs. I think it would be best to have a short `<script>p5._throwValidationErrors = true;</script>` in the HTML immediately after the p5 loading script?